src/Core/Commander/Providers/WMS_Provider.js

```diff
@@ -90,130 +76,52 @@ WMS_Provider.prototype.tileInsideLimit = function tileInsideLimit(tile, layer) {
         return result;
     });
 };
 
-WMS_Provider.prototype.getXbilTexture = function getXbilTexture(tile, layer, bbox, pitch) {
-    var url = this.url(bbox, layer);
-
-    // TODO: this is not optimal: if called again before the IoDriver resolves, it'll load the XBIL again
-    var textureCache = this.cache.getRessource(url);
-
-    if (textureCache !== undefined) {
-        return Promise.resolve({
-            pitch,
-            texture: textureCache.texture,
-            min: textureCache.min,
-            max: textureCache.max,
-        });
-    }
-
-    // bug #74
-    // var limits = layer.tileMatrixSetLimits[coWMTS.zoom];
-    // if (!limits || !coWMTS.isInside(limits)) {
-    //     var texture = -1;
-    //     this.cache.addRessource(url, texture);
-    //     return Promise.resolve(texture);
-    // }
-    // -> bug #74
-    return this._IoDriver.read(url).then((result) => {
-        result.texture = this.getTextureFloat(result.floatArray);
-        result.texture.generateMipmaps = false;
-        result.texture.magFilter = THREE.LinearFilter;
-        result.texture.minFilter = THREE.LinearFilter;
-        result.pitch = pitch;
-
-        // In RGBA elevation texture LinearFilter give some errors with nodata value.
-        // need to rewrite sample function in shader
-        this.cache.addRessource(url, result);
-
-        return result;
-    });
+WMS_Provider.prototype.getXbilTexture = function getXbilTexture(tile, layer) {
+    const url = this.url(tile.bbox.as(layer.projection), layer);
+    return this.getXBilTextureByUrl(url, new THREE.Vector3(0, 0, 1));
 };
 
 WMS_Provider.prototype.executeCommand = function executeCommand(command) {
-    var layer = command.layer;
-    var tile = command.requester;
-    var ancestor = command.ancestor;
+    const parentTextures = command.parentTextures;
+    const tile = command.requester;
 
-    var supportedFormats = {
+    if (parentTextures) {
+        const texture = parentTextures[0];
+        const pitch = TiledImageTools.WMS_WGS84Parent(tile.bbox, texture.bbox);
+        return Promise.resolve({ pitch, texture });
+    }
+
+    const layer = command.layer;
+    const supportedFormats = {
         'image/png': this.getColorTexture.bind(this),
         'image/jpg': this.getColorTexture.bind(this),
         'image/jpeg': this.getColorTexture.bind(this),
         'image/x-bil;bits=32': this.getXbilTexture.bind(this),
     };
 
-    var func = supportedFormats[layer.format];
-    if (func) {
-        var pitch = ancestor ?
-            this.projection.WMS_WGS84Parent(tile.bbox, ancestor.bbox) :
-            new THREE.Vector3(0, 0, 1);
-
-        var bbox = ancestor ?
-            ancestor.bbox :
-            tile.bbox;
-
+    const func = supportedFormats[layer.format];
 
-        return func(tile, layer, bbox, pitch);
+    if (func) {
+        return func(tile, layer);
     } else {
         return Promise.reject(new Error(`Unsupported mimetype ${layer.format}`));
     }
```

Comment:
Note that the elevation texture special case handling doesn't really belong to this provider.
For instance, I might have a WMS server supplying grayscale image to use as a MNT (and I've done this at least twice for demos built with itowns).

So deducing image usage from image format is wrong. It does'nt require fixing here, but keep this in mind while refactoring.